src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -228,14 +228,14 @@ function MoneyRequestConfirmationList({
     const customUnitRateID = TransactionUtils.getRateID(transaction) ?? '-1';
 
     useEffect(() => {
-        if (customUnitRateID || !canUseP2PDistanceRequests) {
+        if ((customUnitRateID && customUnitRateID !== '-1') || !isDistanceRequest) {
             return;
         }
-        if (!customUnitRateID) {
-            const rateID = lastSelectedDistanceRates?.[policy?.id ?? ''] ?? defaultMileageRate?.customUnitRateID ?? '';
-            IOU.setCustomUnitRateID(transactionID, rateID);
-        }
-    }, [defaultMileageRate, customUnitRateID, lastSelectedDistanceRates, policy?.id, canUseP2PDistanceRequests, transactionID]);
+
+        const defaultRate = defaultMileageRate?.customUnitRateID ?? '';
+        const rateID = canUseP2PDistanceRequests ? lastSelectedDistanceRates?.[policy?.id ?? ''] ?? defaultRate : defaultRate;
```

Comment:
I find this to be quite a lot of question marks to follow in one line. I think something like this is easier to follow. 
```suggestion
        const defaultRate = defaultMileageRate?.customUnitRateID ?? '';
        const lastSelectedRate = lastSelectedDistanceRates?.[policy?.id ?? ''] ?? defaultRate;
        const rateID = canUseP2PDistanceRequests ? lastSelectedRate : defaultRate;
```